corehq/apps/users/models.py

```diff
@@ -2334,6 +2331,7 @@ def get_fixture_statuses(user_id):
 class WebUser(CouchUser, MultiMembershipMixin, CommCareMobileContactMixin):
     program_id = StringProperty()
     last_password_set = DateTimeProperty(default=datetime(year=1900, month=1, day=1))
+    domains_accessed = DictProperty()
```

Comment:
You could also store this information as a `DateTimeProperty` on the `DomainMembership`, since it's an attribute of the user's relationship to the domain.

Reply:
good idea, WebUser/CouchUser has plenty of properties already

Reply:
I do agree that it would make more sense to keep that information in `DomainMembership`. As for when we get that info for the report (https://github.com/dimagi/commcare-hq/blob/4cdf4a1309c31e56a547a0fe21ec5d4fe4275189/corehq/apps/accounting/enterprise.py#L142), is there a clean way of getting the right object in `user.domain_memberships`?

Reply:
Yup, `user.get_domain_membership(domain)`.  The user models have a pretty nasty class hierarchy, but it should be in there from one of the mixins, IIRC.

Reply:
Oh, Jenny mentioned superusers - are they in scope?  If so, this approach won't work, as they only have `DomainMembership`s for domains where they've been added.  Probably makes sense to exclude superusers otherwise anyways, since they won't show up in the UI for partners, and things could get weird with support accessing hundreds of projects.

Reply:
Superusers will only be included in the report if they have been added as web users to the domain. Is that what you mean?

Reply:
Perfect, that is what I was thinking of.  Sounds great.